tests/next_tests/integration_tests/feature_tests/iterator_tests/test_scan.py

```diff
@@ -46,7 +46,7 @@ def shifted(inp):
 
     @fundef
     def wrapped(inp):
-        return scan(sum, True, 0.0)(inp, lift(shifted)(inp))
+        return scan(sum, True, 0.0)(inp, shift(Koff, 1)(inp))
```

Comment:
This change breaks something else, if we undo we could merge.

Reply:
I addressed this issue by adding another inline_into_scan pass after temporaries are created

Reply:
We concluded that this is fine for now even though we are not sure whether the ordering of the passes makes sense anymore. This will be addressed as a whole in another PR.